playbooks/common/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml

```diff
@@ -56,3 +56,12 @@
   - import_role:
       name: openshift_web_console
       tasks_from: remove_old_asset_config
+
+- hosts: oo_first_master
+  tasks:
+  - import_role:
+      name: openshift_hosted
+      tasks_from: remove_legacy_env_variables.yml
+  - import_role:
+      name: openshift_hosted
+      tasks_from: registry_service_account.yml
```

Comment:
does order matter here?  we'd want to create/grant the registry SA before modifying the registry DC, ideally.

Reply:
I do not think so. I did it only because of `oo_first_master`.

Reply:
So the order the tasks are defined in does not determine the order they are executed in?

Reply:
@bparees I'm not sure I understand the question.

Reply:
When the upgrade playbook is run, which step will run first:

1) removing the legacy env vars
2) adding/updating the registry service account?

Based on the ordering in the file, I would assume the env variable removing would happen first, and I am suggesting it should be the other way around.  But I am probably just not understanding how ansible works.

Reply:
Because if you remove the env vars first (which does restart the pod) and the registry SA doesn't have the right permissions yet, then the registry pod will come up and not have the right permissions to function, until the second task executes to grant them.

it is a small window but it would be better not to have it (especially in case something fails and the second task doesn't get run correctly)

Reply:
@bparees Fixed.